src/server/util/aws.ts

```diff
@@ -0,0 +1,41 @@
+import { parse, URL } from 'url'
+import { S3 } from 'aws-sdk'
+import { s3Bucket } from '../config'
+
+export const s3 = new S3()
+
+/**
+ * Reformat the pre-signed url to one that will be accepted by S3.
+ *
+ * This is necessary because we used a gov.sg CNAME and configured S3
+ * to only accept requests that have that gov.sg CNAME as the path.
+ *
+ * Broadly speaking, we need to change https://s3.amazonaws.com/file.go.gov.sg/link?search
+ * to https://file.go.gov.sg.s3.amazonaws.com/link?search.
+ */
+const reformatPresignedUrl = (url: string, fileName: string) => {
+  const urlObj = parse(url)
+  const {
+    host,
+    pathname,
+    protocol,
+    search,
+  } = urlObj
+  const newUrl = new URL('https://lorem-ipsum.com')
+  newUrl.protocol = protocol as string
+  newUrl.host = `${pathname?.split('/')[1]}.${host}`
+  newUrl.pathname = fileName
+  newUrl.search = search as string
+  return newUrl.href
+}
+
+export const generatePresignedUrl = async (fileName: string, fileType: string) => {
+  const params = {
+    Bucket: s3Bucket,
+    Key: fileName,
+    ContentType: fileType,
+    Expires: 60, // 60 seconds.
```

Comment:
Does this restrict the duration of uploads to 60s, or is it only the beginning of the upload?

Reply:
The latter; as long as uploading begins before the expiration time, the whole upload process will succeed. [source](https://docs.aws.amazon.com/en_us/AmazonCloudFront/latest/DeveloperGuide/private-content-signed-urls.html#private-content-check-expiration)